File: scoring/src/main/scala/ai/h2o/sparkling/ml/params/HasOutputCol.scala
package ai.h2o.sparkling.ml.params

import org.apache.spark.ml.param.{Param, Params}

trait HasOutputCol extends Params {

  private val outputCol: Param[String] = new Param[String](this, "outputCol", "Output column name")

  setDefault(outputCol, uid + "__output")

  def getOutputCol(): String = $(outputCol)

  def setOutputCol(name: String): this.type = set(outputCol -> name)
}

File: py-scoring/src/ai/h2o/sparkling/ml/params/HasOutputCol.py
from ai.h2o.sparkling.ml.params.H2OTypeConverters import H2OTypeConverters
from pyspark.ml.param import *


class HasOutputCol(Params):
    outputCol = Param(
        Params._dummy(),
        "outputCol",
        "Output columns name",
        H2OTypeConverters.toNullableString())

    def getOutputCol(self):
        return self.getOrDefault(self.outputCol)

    def setOutputCol(self, value):
        return self._set(outputCol=value)
